test/e2e/specs/tools/social-connections__tumblr.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @group jetpack-wpcom-integration
+ */
+
+import {
+	envToFeatureKey,
+	getTestAccountByFeature,
+	envVariables,
+	DataHelper,
+	MarketingPage,
+	RestAPIClient,
+	SidebarComponent,
+	TestAccount,
+	SecretsManager,
+} from '@automattic/calypso-e2e';
+import { Page, Browser } from 'playwright';
+
+declare const browser: Browser;
+
+/**
+ * Sets up a Tumblr social connection for the site.
+ *
+ * Keywords: Social Connections, Marketing, Jetpack, Tumblr, Publicize
+ */
+describe( DataHelper.createSuiteTitle( 'Social Connections: Set up Tumblr' ), function () {
+	let page: Page;
+	let popup: Page;
+
+	let testAccount: TestAccount;
+	let restAPIClient: RestAPIClient;
+	let marketingPage: MarketingPage;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+
+		const features = envToFeatureKey( envVariables );
+		const accountName = getTestAccountByFeature( features );
+		testAccount = new TestAccount( accountName );
+		await testAccount.authenticate( page );
+
+		restAPIClient = new RestAPIClient( testAccount.credentials );
+
+		// Check whether a Tumblr connection exists.
+		const connections = await restAPIClient.getAllPublicizeConnections(
+			testAccount.credentials.testSites?.primary.id as number
+		);
+
+		// If it does, remove the connection.
+		Object.values( connections ).forEach( async ( connection ) => {
+			if ( connection.label === 'Tumblr' ) {
+				console.info( `Removing existing connection for Tumblr for accountName ${ accountName }.` );
+				await restAPIClient.deletePublicizeConnection( connection.site_ID, connection.ID );
+			}
+		} );
```

Comment:
**Suggestion (blocking):** As a heads up, it's usually a bad idea to mix `forEach` with `async-await`! It doesn't work how you might expect. I just got burned by this yesterday haha. 😆 For this specific case, you could do a `for... of` on the `connections`.

[Helpful reference](https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971)

Reply:
Thanks for the heads up! I always get confused on how best to iterate over an array and perform async action on each in JS/TS.